### PR TITLE
manually set TO&E force commander

### PR DESCRIPTION
**Pull request by NickAragua**
Implements  #3540 by allowing the manual selection of a force commander from the TO&E menu if there's more than one "highest rank". Automatically updates the force commander if a unit/person is added/removed from the force/campaign.

### CODE REVIEW

File: MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java [hunk 1]
```diff
@@ -899,6 +908,23 @@ protected Optional<JPopupMenu> createPopupMenu() {
                 }
                 JMenuHelpers.addMenuIfNonEmpty(menu, unsorted);
                 JMenuHelpers.addMenuIfNonEmpty(popup, menu);
+                
+                // still in the multiple selection block
+                List<UUID> eligibleCommanders = force.getEligibleCommanders(gui.getCampaign());
+                if (eligibleCommanders.size() > 0) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java:914]:
```suggestion
                if (!eligibleCommanders.isEmpty()) {                
```

**Review by Windchild292 — COMMENTED** [review]
One improvement, otherwise this looks god

**Comment by SJuliez** [comment]
I've committed the suggestion which seemed valid. Hope I didnt destroy anything.

**Review by SJuliez — APPROVED** [review]

**Review by Windchild292 — APPROVED** [review]

**Comment by NickAragua** [comment]
Still waiting on uploaded save from bug report

**Comment by Thom293** [comment]
> Still waiting on uploaded save from bug report

Did you mean mine?  Sorry did you need something else?  

https://github.com/MegaMek/mekhq/issues/3781

**Comment by Thom293** [comment]
Sorry I missed the fix posted earlier.  Tested it just now.  

Seems to work great.  If someone set as commander is moved from a lance, their command flag is wiped and the next highest rank is then listed as commander.  Can select any equals as commander, move them, and it causes no more issues. Looks good to me.

**Pull request merged**